flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -260,4 +268,57 @@ public Optional<OperationFlowsView> getLastFlowOperationByResourceCrn(String res
         List<FlowLog> flowLogs = flowLogDBService.getLatestFlowLogsByCrnInFlowChain(resourceCrn);
         return flowOperationStatisticsService.createOperationResponse(resourceCrn, flowLogs);
     }
+
+    public List<FlowLogResponse> getFlowLogsByIds(List<String> flowIds, Pageable pageable) {
+        LOGGER.info("Getting flow logs by flow ids list {}", flowIds);
+        return flowLogDBService.getFlowLogsByFlowIdsCreatedDesc(new HashSet<>(flowIds), pageable)
+                .stream().map(flowLog -> flowLogConverter.convert(flowLog)).collect(toList());
+    }
+
+    public List<FlowCheckResponse> getFlowChainsByChainIds(List<String> chainIds, Pageable pageable) {
```

Comment:
in overall I find it hard to follow whats going on here, even with the comments with should be removed by the way, and if necessary this should be splitted into smaller methods instead of comments

Reply:
i will remove the comments.

Reply:
split the method and added additional test case to verify if the child chains are being properly processed and added to the flow chain check response